### DISCUSSION TITLE
What's Your Scrubbing and Playback Performance Like?

### DISCUSSION THREAD

**Discussion by tangierc** [latenitefilms/BRAWToolbox#120] [General]
Folks, I am just curious what your performance is like using BRAW Toolbox? I've noticed recently that BRAW Toolbox and FCP takes  big performance hit compared to using the braw clips in Resolve using the same USB-C 3.1 Gen 2 source drive for the media. Perhaps that's to be expected since they're generators. I am curious if anyone has tried assessing performance differences between FCP and Resolve using the same source clips from the same drive. I noticed also that if my FCP projects are long with a lot of BRAW Toolbox generated clips the playback and scrubbing performance also take a big hit and is choppy in FCP. All of my clips are at eighth decode quality. I may have to get a Thunderbolt drive to compensate for the performance.

M2 Max MBP / Ventura 13.3.1 / FCP 10.6.5 / BRAW Toolbox 1.08

**Comment by latenitefilms** [comment]
Due to limitations with the [FxPlug4 API](https://developer.apple.com/documentation/professional_video_applications/fxplug?language=objc) - BRAW Toolbox works a bit differently than DaVinci Resolve, despite them both using the same Blackmagic RAW SDK. 

BRAW Toolbox currently only processes one frame at a time, from start to finish, whereas on Resolve, depending on the system and number of GPUs, you can have multiple frames in flight at the same time (i.e. it might be reading frame 3, whilst decoding frame 2 and processing frame 1).

I'm constantly trying to come up with tricks and workarounds to make things faster, and working with Apple and Blackmagic to make this happen.

However, FxPlug4 was always designed as an "effects" platform - so we're making it do something it was never designed for.

It's also worth noting also that Final Cut Pro also has performance issues when you have very complex libraries - regardless of whether those clips are ProRes or BRAW Toolbox clips. As more general advice, it's always a good idea to break things up into multiple events, and if you're working on a feature film, keep your latest edit in an event of its own.

**Reply by latenitefilms** [reply]
It's also worth nothing that granting sandbox access does have a very minor performance improvement, because if Final Cut Pro/BRAW Toolbox has sandbox access to the drive that contains your BRAW clips, then it doesn't have to resolve the security-scoped bookmarks for each frame - it can just use the file path.

See: https://brawtoolbox.io/collaboration/
